diff --git a/rules.mk b/rules.mk
index 92f34db68..08cf2d6c3 100644
--- a/rules.mk
+++ b/rules.mk
@@ -27,8 +27,11 @@
 	--cfg 'libc_ptr_addr_of' \
 	--cfg 'libc_underscore_const_names' \
 	--cfg 'libc_union' \
+	-A unknown-lints \

 MODULE_LIBRARY_DEPS := \
-	external/rust/crates/rustc-std-workspace-core \
+	trusty/user/base/lib/libcompiler_builtins-rust \
+	trusty/user/base/lib/libcore-rust \
+	trusty/user/base/lib/libc-trusty \

